actions/event_calendar/upgrade.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Event Calendar upgrade action
+ */
+
+$plugins_path = elgg_get_plugins_path();
+
+require_once "{$plugins_path}event_calendar/version.php";
+
+$local_version = elgg_get_plugin_setting('version', 'event_calendar');
+
+if ($version <= $local_version) {
+	register_error(elgg_echo('event_calendar:upgrade:no_updates'));
```

Comment:
I don't think a system_message would be correct. The upgrade button gets only displayed if a new upgrade script is available. If the admin executes then the upgrade and the version check says there's no upgrade available to be done something has gone wrong for sure. Or if an admin calls the action by other means (even if the upgrade button is not shown) and there's (as to be expected) no upgrade found to be executed it's also rather an error.